src/Listener/RegexListener.php

```diff
@@ -1,269 +1,296 @@
 <?php
+
 declare(strict_types=1);
 
 namespace JsonStreamingParser\Listener;
 
 /**
  * Allow to select elements from a JSON being loaded via a regex expression.
  * As the JSON is loaded you can use expressions such as /name to select
- * a name attribute from the root of the document.  
- * For arrays, /1/name which is the name element from the second element 
+ * a name attribute from the root of the document.
+ * For arrays, /1/name which is the name element from the second element
  * of an array. Using things like /\d* will match any array element.
- * 
+ *
  * Basic usage (with file tests/data/example.json):
  * $listener = new RegexListener(["/1/name" => function ($data ) {
  *     echo "/1/name=".$data.PHP_EOL;
  * }]);
  * $parser = new Parser(fopen($filename, 'rb'), $listener);
  * $parser->parse();
- * 
+ *
  * Capture groups can also be used to provide the element path to the data
  * located, so (/\d*) will pass the array element number along with the
  * element value.
- * 
+ *
  * Example using capture groups:
  * $listener = new RegexListener(["(/\d*)" => function ($data, $path ) {
  *     echo $path."=".$data['name'].PHP_EOL;
  * }]);
  * $parser = new Parser(fopen($filename, 'rb'), $listener);
  * $parser->parse();
- * 
+ *
  * It's also possible to stop the load if you only need specific data.
- * 
+ *
  * Example to stop read (with file tests/data/ratherBig.json):
  * $listener = new RegexListener();
  * $parser = new Parser(fopen($filename, 'rb'), $listener);
  * $listener->setMatch(["/total_rows" => function ($data ) use ($parser) {
  *     echo "/total_rows=".$data.PHP_EOL;
  *     $parser->stop();
  * }]);
- * 
+ *
  * @author NigelRen <nigelrel3@yahoo.co.uk>
- * 
  */
-
-class RegexListener implements ListenerInterface  {
+class RegexListener implements ListenerInterface
+{
     /**
      * Constant to flag that it is an object level.
+     *
      * @var int
      */
     const OBJECT = -1;
     /**
      * Constant to flag that it is an array level.
+     *
      * @var int
      */
     const ARRAY = 0;
-    
+
```

Comment:
@MAXakaWIZARD example I believe.

Reply:
this one is removing extra whitespace at the line end